apps/mobile/src/features/activity/utils/format-activity.ts

```diff
@@ -1,52 +1,134 @@
 import { t } from '@lingui/macro';
+import dayjs from 'dayjs';
 
-import { ActivityType } from '@leather.io/models';
-
-export function formatActivityType(activityType: ActivityType) {
-  return {
-    deploySmartContract: t({
-      id: 'activity.type.deploySmartContract',
-      message: 'Deploy Smart Contract',
-    }),
-    executeSmartContract: t({
-      id: 'activity.type.executeSmartContract',
-      message: 'Execute Smart Contract',
-    }),
-    lockAsset: t({
-      id: 'activity.type.lockAsset',
-      message: 'Lock Asset',
-    }),
-    sendAsset: t({
-      id: 'activity.type.sendAsset',
-      message: 'Send',
-    }),
-    receiveAsset: t({
-      id: 'activity.type.receiveAsset',
-      message: 'Receive',
-    }),
-    swapAssets: t({
-      id: 'activity.type.swapAssets',
-      message: 'Swap Assets',
-    }),
-    connectApp: t({
-      id: 'activity.type.connectApp',
-      message: 'Connect App',
-    }),
-    signMessage: t({
-      id: 'activity.type.signMessage',
-      message: 'Sign Message',
-    }),
-    walletAdded: t({
-      id: 'activity.type.walletAdded',
-      message: 'Wallet Added',
-    }),
-    receiveAnnouncement: t({
-      id: 'activity.type.receiveAnnouncement',
-      message: 'Announcement',
-    }),
-    featureWaitlistNotification: t({
-      id: 'activity.type.featureWaitlistNotification',
-      message: 'Waitlist Notification',
-    }),
-  }[activityType];
+import { BaseOnChainActivity, OnChainActivity } from '@leather.io/models';
+
+interface FormatActivityStatusProps {
+  type: OnChainActivity['type'];
+  status: BaseOnChainActivity['status'];
+}
+
+export function formatActivityStatus({ type, status }: FormatActivityStatusProps) {
+  switch (true) {
```

Comment:
I got some pushback about using `switch(true)` in the past but I think it's a good fit here to map out all the status's and text. 

I'm open to refactoring if there is a better way

Reply:
Great idea to use the `map`, I hadn't considered that approach. I'll change it and revise the name

Reply:
I've updated this. FYI there is no such thing as `pending` for `receive` but I have added a placeholder. 

I also added some more activities I found using the map but we don't have a design for yet.